elastic/kibana
# Enable CSV reporting in serverless

## Summary

This PR sets up the reporting plugin for the serverless implementation by adding properties to the existing reporting config. 

Image reporting is enabled for dev mode but disabled for serverless. Canvas is disabled for serverless. 

## To Test

Run `yarn es snapshot --license trial` in one terminal and then `yarn start`. Load sample data and you should be able to see the option to have PDF and PNG reports in Dashboard's Share Menu.

![Screenshot 2023-07-25 at 9 40 30 AM](https://github.com/elastic/kibana/assets/20343860/c258a14d-6cc7-4fdf-9bb1-4dc3b15d371b)

Now run `yarn es snapshot --license trial` and `yarn serverless-es`. You should see that Dashboard's share menu does not include PDF or PNG Reports. However there is still the option to see run CSV reports and see the Reporting in Management.

![Screenshot 2023-07-25 at 9 42 16 AM](https://github.com/elastic/kibana/assets/20343860/638691dc-6c2f-41ed-a8d3-d5d38c15fa91)


### Checklist
